test/unit/lodash.spec.js

```diff
@@ -0,0 +1,1326 @@
+                });
+
+                expect(_.findKey.apply(_, datasets.findKey4(obj1))).to.eql(_.findKey(obj1, params[1]));
+            });
+
+        });
+    });
+
+
+    describe('findLastKey', function () {
+        describe('findLastKey1', function () {
+            var params;
+            before(function(){
+                params = datasets.findLastKey1();
+            });
+            it('should check types', function(){
+                var element2 = params[1];
+                expect(element2 instanceof Function).to.eql(true);
+            });
+            it('should return key of last element matching given criteria', function(){
+                var randomBool = function(){
+                    //function _.random Returns random true or false
+                    return _.random() < 0.5;
+                };
+
+                var orders  = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index){
+                    orders['order' + index] = {pizza: randomBool(), pasta: randomBool(), lasagne: randomBool()};
+                });
+
+                expect(_.findLastKey.apply(_, datasets.findLastKey1(orders))).to.eql(_.findLastKey(orders, params[1]));
+            });
+        });
+
+        describe('findLastKey2', function () {
+            var params;
+            before(function(){
+                params = datasets.findLastKey2();
+            });
+            it('should check types', function(){
+                var elem2 = params[1];
+                expect(typeof elem2).to.eql('object');
+            });
+            it('should return new object filtred thru given obj', function(){
+                var randomBool = function(){
+                    //function _.random Returns random true or false
+                    return _.random() < 0.5;
+                };
+
+                var orders  = {};
+
+                //function _.times Returns the array. In this case 100 results.
+                _.times(100, function(index){
+                    orders['order' + index] = {pizza: randomBool(), pasta: randomBool(), lasagne: randomBool()};
+                });
+
+                expect(_.findLastKey.apply(_, datasets.findLastKey2(orders))).to.eql(_.findLastKey(orders, params[1]));
+            });
+        });
+
+        describe('findLastKey3', function () {
+            var params;
+            before(function(){
+                params = datasets.findLastKey3();
+            });
+            it('should check types', function(){
+                var elem2 = params[1];
+
+                expect(elem2 instanceof Array).to.eql(true);
+                expect(params).to.have.length(2);
+            });
+            it('should return object filtred thru given array', function(){
+                var randomBool = function(){
+                    //function _.random Returns random true or false
+                    return _.random() < 0.5;
```

Comment:
being more accurate .random will return 0 or 1 but result of comparision will be true or false, You could also do it with !_.random() or !!_.random()